ext/jsonnet/callbacks.c

```diff
@@ -123,14 +129,26 @@ import_callback_entrypoint(void *ctx, const char *base, const char *rel, char **
 
     if (state) {
 	VALUE msg = rescue_callback(state, "cannot import %s from %s", rel, base);
+#ifdef HAVE_JSONNET_IMPORT_CALLBACK_0_19
+	*buf = rubyjsonnet_str_to_cstr(vm->vm, msg);
+	*buflen = RSTRING_LEN(msg);
+	return 1;
+#else
 	*success = 0;
 	return rubyjsonnet_str_to_cstr(vm->vm, msg);
+#endif
     }
 
     result = rb_Array(result);
-    *success = 1;
     *found_here = rubyjsonnet_str_to_cstr(vm->vm, rb_ary_entry(result, 1));
+#ifdef HAVE_JSONNET_IMPORT_CALLBACK_0_19
+    *buf = rubyjsonnet_str_to_cstr(vm->vm, rb_ary_entry(result, 0));
```

Comment:
I think we are expected to return arbitrary bytes instead of c string (null terminated) with `buf` because `importbin` may import data containing null bytes, so we may need another helper function like below.

```c
char *
rubyjsonnet_str_to_ptr(struct JsonnetVm *vm, VALUE str)
{
    const char *const ptr = StringValuePtr(str);
    char *const buf = jsonnet_realloc(vm, NULL, RSTRING_LEN(str));
    memcpy(buf, ptr, RSTRING_LEN(str));
    return buf;
}
```

https://github.com/google/jsonnet/blob/ca2d672ffe4c243570671ee0cd62d887f123372e/include/libjsonnet.h#L72